### PR TITLE
fix: solve permission denied when running with act

### PR DESCRIPTION
**Pull request by pietrobolcato**
When testing the action locally using [act](https://github.com/nektos/act), it returns

`/var/run/act/workflow/1-composite-0.sh: line 2: /var/run/act/actions/snok-install-poetry@v1.3.3/main.sh: Permission denied`

This commit fixes the issue by setting the execution permission on the main installer script

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: "Install Poetry Action"
+name: "Install Poetry Action - act fix"
```
Comment (miigotu) [inline action.yml:1]: You'll have to undo this change

**Comment by JonasKs** [comment]
I'm not sure why this change would be needed in act, when it is not required in GitHub actions? To my knowledge act uses the same docker images, so whether the file has execution rights or not, should result in the same in both environments? 

To me this sounds like an `act` bug.

**Comment by sondrelg** [comment]
Thanks for the PR :slightly_smiling_face: 

If the script isn't executable, would it be enough to `chmod` it locally and commit it? I can test with `act` after work today.

Looks like we also have some existing tests failing, but that seems unrelated.

**Comment by JonasKs** [comment]
Good suggestion! Seems like it's [possible](https://stackoverflow.com/a/40979016)

**Comment by pietrobolcato** [comment]
Great find! Let us know once you tried it out, thank you!

**Comment by deronnax** [comment]
I think it can be closed now, can't it? Anyway if it works on Github Actions and not in act, it's totally a bug in act, and that need to be fixed in act.

**Comment by sondrelg** [comment]
Yep think you're right
